### PR TITLE
[FEATURE] Delete regression test

### PR DESCRIPTION
**Pull request by T1duS**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Added option to delete regression tests along with confirmation and unit tests
https://github.com/CCExtractor/sample-platform/pull/235 v2

### CODE REVIEW

**Comment by satyammittal** [comment]
@T1duS ur travis test failed. Please check. https://travis-ci.org/CCExtractor/sample-platform/jobs/447074091

File: mod_regression/controllers.py [hunk 1]
```diff
@@ -62,9 +62,39 @@ def test_view(regression_id):
 
 
 @mod_regression.route('/test/<regression_id>/delete')
+@template_renderer()
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: Why use a template renderer if you don't use it for anything?

File: tests/test_regression/TestControllers.py [hunk 2]
```diff
@@ -30,3 +30,18 @@ def test_regression_test_status_toggle(self):
                 self.assertEqual('False', response.json['active'])
             else:
                 self.assertEqual('True', response.json['active'])
+                
+    def test_regression_test_delete(self):
```
Comment (canihavesomecoffee) [inline tests/test_regression/TestControllers.py]: Tests should be split up in different cases.

File: mod_regression/controllers.py [hunk 3]
```diff
@@ -62,9 +62,39 @@ def test_view(regression_id):
...
+        abort(404) # aborting if a regression test with the given id doesn't exist
+
+    customized_tests = CustomizedTest.query.filter(CustomizedTest.regression_id == regression_id).all() # finding all customized tests that have the given regression test
+    regression_test_outputs = RegressionTestOutput.query.filter(RegressionTestOutput.regression_id == regression_id).all() # finding all regression test outputs that have the given regression test
+    test_results = TestResult.query.filter(TestResult.regression_test_id == regression_id).all() # finding all test results that have the given regression test
+    test_result_files = TestResultFile.query.filter(TestResultFile.regression_test_id == regression_id).all() # finding all test result files that have the given regression test
+
+    # Deleting the tests/files which have the given regression test
+
+    for customized_test in customized_tests:
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: Check the models to see if this needs to be done explicitely, or if it's implicitely deleted by the relational models.

**Comment by canihavesomecoffee** [comment]
Merged #236.

**Pull request closed without merging**